src/server/scripts/Commands/cs_anticheat.cpp

```diff
@@ -0,0 +1,210 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Language.h"
+#include "ScriptMgr.h"
+#include "ObjectMgr.h"
+#include "Chat.h"
+#include "SpellAuras.h"
+#include "AnticheatMgr.h"
+
+class anticheat_commandscript : public CommandScript
+{
+public:
+    anticheat_commandscript() : CommandScript("anticheat_commandscript") { }
+
+    std::vector<ChatCommand> GetCommands() const override
+    {
+        static std::vector<ChatCommand> anticheatCommandTable =
+        {
+            { "global", rbac::RBAC_PERM_COMMAND_ANTICHEAT_GLOBAL, true,  &HandleAntiCheatGlobalCommand, "" },
+            { "player", rbac::RBAC_PERM_COMMAND_ANTICHEAT_PLAYER, true,  &HandleAntiCheatPlayerCommand, "" },
+            { "handle", rbac::RBAC_PERM_COMMAND_ANTICHEAT_HANDLE, true,  &HandleAntiCheatHandleCommand, "" },
+            { "jail",   rbac::RBAC_PERM_COMMAND_ANTICHEAT_JAIL,   false, &HandleAnticheatJailCommand,   "" },
+            { "warn",   rbac::RBAC_PERM_COMMAND_ANTICHEAT_WARN,   true,  &HandleAnticheatWarnCommand,   "" }
+        };
+        static std::vector<ChatCommand> commandTable =
+        {
+            { "anticheat", rbac::RBAC_PERM_COMMAND_ANTICHEAT, true, nullptr, "", anticheatCommandTable }
+        };
+        return commandTable;
+    }
+
+    static bool HandleAnticheatWarnCommand(ChatHandler* handler, char const* args)
+    {
+        if (!sWorld->getBoolConfig(CONFIG_ANTICHEAT_ENABLE))
+        {
+            handler->SendSysMessage(LANG_COMMAND_ACDISABLED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        Player* pTarget = nullptr;
+        char* command = strtok(const_cast<char*>(args), " ");
+
+        if (command)
+        {
+            std::string strCommand(command);
+            normalizePlayerName(strCommand);
+
+            pTarget = ObjectAccessor::FindPlayerByName(strCommand); //get player by name
+        }
+        else
+            pTarget = handler->getSelectedPlayer();
+
+        if (!pTarget)
+        {
+            handler->SendSysMessage(LANG_PLAYER_NOT_FOUND);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        ChatHandler(pTarget->GetSession()).SendSysMessage(LANG_COMMAND_ACWARN);
+        return true;
+    }
+
+    static bool HandleAnticheatJailCommand(ChatHandler* handler, char const* args)
+    {
+        if (!sWorld->getBoolConfig(CONFIG_ANTICHEAT_ENABLE))
+        {
+            handler->SendSysMessage(LANG_COMMAND_ACDISABLED);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        static WorldLocation const jailPos(1, 16226.5f, 16403.6f, -64.5f, 3.2f);
+
+        Player* pTarget = nullptr;
+        char* command = strtok(const_cast<char*>(args), " ");
```

Comment:
Not really, as Rochet2 pointed out in that link, const-ness is casted away as it is here (except the cast there is not explicit, rather a C-style cast, this is why I prefer const_cast, because it's explicit of my intention)